src/malli/instrument/cljs.clj

```diff
@@ -0,0 +1,156 @@
+(ns malli.instrument.cljs
+  (:require
+    [cljs.analyzer.api :as ana-api]
+    [malli.core :as m]
+    [malli.generator :as mg]))
+
+;;
+;; instrument
+;;
+
+(defn emit-instrument-fn [{:keys [gen filters] :as instrument-opts} {:keys [schema] :as schema-map} ns-sym fn-sym]
+  ;; gen is a function
+  (let [schema-map (assoc (select-keys schema-map [:gen :scope :report])
+                     ;; At macroexpansion time the schema will be a JVM object, this converts it to a JS object.
+                     :schema `(m/function-schema ~(m/form schema)))
+        schema-map-with-gen
+                   (as-> (merge (select-keys instrument-opts [:scope :report :gen]) schema-map) $
+                     ;; use the passed in gen fn to generate a value
+                     (cond (and gen (true? (:gen schema-map))) (assoc $ :gen gen)
+                           :else (dissoc $ :gen)))
+        replace-var-code
+                   `(do
+                      (swap! instrumented-vars #(assoc % '~fn-sym ~fn-sym))
+                      (set! ~fn-sym (m/-instrument ~schema-map-with-gen ~fn-sym))
+                      (.log js/console "..instrumented" '~fn-sym)
+                      '~fn-sym)]
+    (if filters
+      `(when (some #(% '~ns-sym (var ~fn-sym) ~schema-map) ~filters)
+         ~replace-var-code)
+      replace-var-code)))
+
+(defn instrument*
+  [{:keys [data] :or {data (m/function-schemas)} :as opts}]
+  (let [r
+        (reduce
+          (fn [acc [ns-sym sym-map]]
+            (reduce-kv
+              (fn [acc' fn-sym schema-map]
+                (conj acc'
+                  (emit-instrument-fn opts schema-map ns-sym (symbol (str ns-sym) (str fn-sym)))))
+              acc sym-map))
+          []
+          data)]
+    `(filterv some? ~r)))
+
+(defmacro instrument!
+  "Applies instrumentation for a filtered set of function Vars (e.g. `defn`s).
+   See [[malli.core/-instrument]] for possible options."
+  ([] (instrument* {}))
+  ([opts] (instrument* opts)))
+
+;;
+;; unstrument
+;;
+
+(defn emit-unstrument-fn [{:keys [schema filters] :as opts} ns-sym fn-sym]
+  (let [opts (assoc (select-keys opts [:gen :scope :report])
+               ;; At macroexpansion time the schema will be a JVM object, this converts it to a JS object.
+               :schema `(m/function-schema ~(m/form schema)))
```

Comment:
here too.

Reply:
calling `m/form` also removes the possibility to use non-registered Schema Instances:

```clj
(def MyInt (m/-simple-schema {:type 'MyInt, :pred int?}))

(m/validate MyInt 2) ; => true

(m/form MyInt) ; => MyInt

;; there is no way to construct the local instance from form.
(m/schema (m/form MyInt))
; =throws=> :malli.core/invalid-schema {:schema MyInt}
```

I'm testing if we can just use `schema` in the three places.

Reply:
with my skills, we can't.

Reply:
ahh that's unfortunate! I will use these examples to see if I can get it working. I _think_ moving everything to happen at runtime (in cljs) should be possible. I will report back when I have an update.

Reply:
awesome, looking forward to it!